### PR TITLE
fix: DE41706: Sync item rendering.

### PR DESCRIPTION
**Pull request by JasperAtD2L**
https://rally1.rallydev.com/#/357251704080d/workviews?detail=%2Fdefect%2F464034359184&view=b064b584-6337-43c1-b9a9-28047568284d&fdp=true?fdp=true

https://rally1.rallydev.com/#/357251704080d/custom/367300408400?detail=%2Fdefect%2F483995040488&fdp=true?fdp=true

Render all items only once the item details have been fetched.

### CODE REVIEW

File: components/d2l-work-to-do/d2l-work-to-do.js [hunk 1]
```diff
@@ -381,13 +389,18 @@ class WorkToDoWidget extends EntityMixinLit(LocalizeWorkToDoMixin(LitElement)) {
...
 		const detailedSkeleton = html`
 			${immersiveNav()}
-			<d2l-work-to-do-activity-list-header skeleton fullscreen></d2l-work-to-do-activity-list-header>
-			<d2l-list separators="none">
-				<d2l-work-to-do-activity-list-item-detailed skeleton href=' ' token=' '></d2l-work-to-do-activity-list-item-detailed>
-				<d2l-work-to-do-activity-list-item-detailed skeleton href=' ' token=' '></d2l-work-to-do-activity-list-item-detailed>
-				<d2l-work-to-do-activity-list-item-detailed skeleton href=' ' token=' '></d2l-work-to-do-activity-list-item-detailed>
-			</d2l-list>
-			<div class="d2l-load-more-button-skeleton d2l-skeletize"></div>
+			<div class="d2l-work-to-do-fullscreen-container">
```
Comment (JasperAtD2L) [inline components/d2l-work-to-do/d2l-work-to-do.js:392]: Not related but noticed the fullscreen skeleton layout was quite different than the final layout. And it was causing quite a bit of jumping between renders.

**Comment by mbardakovD2L** [comment]
how does this work in fullscreen view, when we load more? does it see that we have X number of new activities, create X skeletons, and then flip all X skeletons to actual items once the data is all there?

File: components/d2l-work-to-do/d2l-work-to-do.js [hunk 2]
```diff
@@ -454,6 +467,21 @@ class WorkToDoWidget extends EntityMixinLit(LocalizeWorkToDoMixin(LitElement)) {
 		return 'loading';
 	}
 
+	/**
+	 * Fired when all the data required to render the activity is present.
+	 * Event is guarunteed to fire for success and failures.
```
Comment (mbardakovD2L) [inline components/d2l-work-to-do/d2l-work-to-do.js:472]: nit: guaranteed

**Review by bearfriend — DISMISSED** [review]
I was personally confused by the name `initialLoad`. I thought it would start as `false` and switch to `true`, but not a big deal. It's obvious what it's doing.

**Comment by JasperAtD2L** [comment]
> how does this work in fullscreen view, when we load more? does it see that we have X number of new activities, create X skeletons, and then flip all X skeletons to actual items once the data is all there?

On load more the _initalLoad is false and thus the skeleton attribute will never get re-applied. And the data loaded hander is effectively disabled.

**Review by bearfriend — APPROVED** [review]
🔤

**Comment by mbardakovD2L** [comment]
> > how does this work in fullscreen view, when we load more? does it see that we have X number of new activities, create X skeletons, and then flip all X skeletons to actual items once the data is all there?
> 
> On load more the _initalLoad is false and thus the skeleton attribute will never get re-applied. And the data loaded hander is effectively disabled.

That makes sense - I'm happier with that than with what I described

**Review by mbardakovD2L — APPROVED** [review]

**Pull request merged**